lib/SimpleSAML/XML/Parser.php

```diff
@@ -41,11 +42,7 @@ public static function fromSimpleXMLElement(\SimpleXMLElement $element)
         /* Create a new parser with the xml document where the namespace definitions
          * are added.
          */
-
-        /** @var string $xml */
-        $xml = $element->asXML();
-        $parser = new Parser($xml);
-        return $parser;
+        return new Parser($xml = $element->asXML());
```

Comment:
The problem here is not the return value, but the argument passed to `\SimpleSAML\XML\Parser::__construct()`, which cannot be `false`. `asXML()` can actually return false:

https://www.php.net/manual/en/simplexmlelement.asxml.php

Reply:
I wonder if we should actually check for error. I'm trying to see under what circumstances `false` can be returned by `asXML()`, but I'm failing to find it...

Reply:
The return-value is also an issue.. Psalm will say that 'object' does not match the return-type 'Parser'.
I've added a test for the false-case.. it may happen if the output is an empty string, or if PHP runs out of memory, if I read the PHP source code correctly..